### DISCUSSION TITLE
Can launch configurations be generated at build time?

### DISCUSSION THREAD

**Discussion by RMPers** [microsoft/vscode-cpptools#7712] [Q&A]
Running CMake allows you to put your build output anywhere in the project. Even having multiple output folders is possible, like for instance builds with different compiler options. A build can have multiple targets, and for each target a launch config will have to be specified. Instead of manually adding a launch configuration for every target in every build output folder I'd like to understand if this can be done as a step in my build. Or if there is a better way.
Thx

**Comment by sean-mcmanus** [comment]
I don't know. Have you tried the CMake Tools extension?

**Reply by RMPers** [reply]
There is a proposal in CMake Tools to use command substitution in launch.json. But it does not feel like a solution I want since it would require me to run CMake from within VSCode just to populate VSCode variables.
It would have been nice if path to program didn't have to be specified within the launch.json and instead just be coming file selected in explorer or something.
New to VSCode and not sure if such thing is possible.
